src/test/java/com/github/stickerifier/stickerify/junit/TempFilesCleanupExtension.java

```diff
@@ -0,0 +1,46 @@
+package com.github.stickerifier.stickerify.junit;
+
+import static org.junit.jupiter.api.Assumptions.abort;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Custom JUnit extension used to clear temp files
+ * generated by unit tests during their execution.
+ */
+public class TempFilesCleanupExtension implements AfterAllCallback {
+
+	@Override
+	public void afterAll(ExtensionContext context) throws IOException {
+		deleteTempFiles();
+	}
+
+	private static void deleteTempFiles() throws IOException {
+		try (var files = Files.list(Path.of(System.getProperty("java.io.tmpdir")))) {
+			files.filter(Files::isRegularFile)
+					.map(Path::toFile)
```

Comment:
This map is not needed, since in the delete method you revert back to a Path object. Instead just use path.getName().toString() inside the filter